pkg/prometheus/promcfg.go

```diff
@@ -1547,6 +1548,13 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 		// config as well, make sure to path the right namespace here.
 		k8sSDConfig = addTLStoYaml(k8sSDConfig, "", apiserverConfig.TLSConfig)
 	}
+	if attachMetadata != nil {
```

Comment:
We should make sure to attach this key only when the version is greater or equal 2.35.0. You can see an example of this pattern here: https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go#L192

Reply:
ok.